### PR TITLE
add binding to constant true/false for BoolExpr::And/BoolExpr::Or

### PR DESCRIPTION
**Pull request by AllenZzw**

### CODE REVIEW

File: crates/huub/src/model/bool.rs [hunk 1]
```diff
@@ -204,13 +204,21 @@ impl BoolExpr {
...
 						BoolViewInner::Const(false) => {}
 						BoolViewInner::Const(true) => return bind_const(&mut slv.oracle, true),
-						BoolViewInner::Lit(l) => lits.push(Formula::Atom(l)),
+						BoolViewInner::Lit(l) => {
+							lits.push(Formula::Atom(l));
+							all_false = false;
+						}
 					}
 				}
+				if all_false {
```
Comment (Dekker1) [inline crates/huub/src/model/bool.rs]: Can't we just check `lits.is_empty()`?

File: crates/huub/src/model/bool.rs [hunk 2]
```diff
@@ -222,13 +230,20 @@ impl BoolExpr {
...
 						BoolViewInner::Const(true) => {}
 						BoolViewInner::Const(false) => return bind_const(&mut slv.oracle, false),
-						BoolViewInner::Lit(l) => lits.push(Formula::Atom(l)),
+						BoolViewInner::Lit(l) => {
+							lits.push(Formula::Atom(l));
+							all_true = false;
+						}
 					}
 				}
+				if all_true {
```
Comment (Dekker1) [inline crates/huub/src/model/bool.rs]: Same here.

File: crates/huub/src/model/bool.rs (hunk 1 above)
Comment (AllenZzw) [reply crates/huub/src/model/bool.rs]: Updated, thanks!

File: crates/huub/src/model/bool.rs (hunk 2 above)
Comment (AllenZzw) [reply crates/huub/src/model/bool.rs]: Updated.

**Pull request closed without merging**